server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -776,15 +776,6 @@ void deleteIndexStore(String reason, IndexMetaData metaData, ClusterState cluste
                     throw new IllegalStateException("Can't delete index store for [" + index.getName() +
                         "] - it's still part of the indices service [" + localUUid + "] [" + metaData.getIndexUUID() + "]");
                 }
-
-                if (clusterState.metaData().hasIndex(index.getName()) && (clusterState.nodes().getLocalNode().isMasterNode() == true)) {
```

Comment:
This code looks old and somewhat bogus -- it seems that it would avoid deleting index metadata for indices that still exist in the cluster state _by name_ (not by UUID) on master nodes.

Reply:
Jup seems there was no deeper reasoning behind this back then in https://github.com/elastic/elasticsearch/pull/9985#discussion_r25805657

Reply:
It's great this code can finally go and the code that operates on the data directories is separate from the one that writes out the cluster state.